src/HalfIntegers.jl

```diff
@@ -444,8 +444,8 @@ julia> twice(Complex{BigInt}, HalfInt(5/2) + HalfInt(3)*im)
 5 + 6im
 ```
 """
-twice(T::Type{<:Integer}, x::Number) = T(twice(x))
-twice(T::Type{<:Integer}, x::Integer) = twice(T(x)) # convert to T first to reduce probability of overflow
+twice(T::Type{<:Integer}, x::Number) = convert(T, twice(x))::T
+twice(T::Type{<:Integer}, x::Integer) = twice(convert(T, x)::T) # convert to T first to reduce probability of overflow
```

Comment:
Is the `::T` actually necessary?

Reply:
It's not necessary, but it's a guard against bad implementations to avoid accidental stack-overflows. If you feel that it adds to clutter, I can remove it.

Reply:
No, you don’t have to remove it. I was just curious whether you encountered a case where the `::T` actually helped.